src/core/utils/net.ts

```diff
@@ -103,22 +103,33 @@ export async function validateDevServerConfig(url: string | undefined, timeout:
     const resolvedPortNumber = await isAcceptingTcpConnections({ port, host: hostname });
     if (resolvedPortNumber !== 0) {
       const spinner = ora();
-      try {
-        spinner.start(`Waiting for ${chalk.green(url)} to be ready`);
-        await waitOn({
-          resources: [`tcp:${hostname}:${port}`],
-          delay: 1000, // initial delay in ms, default 0
-          interval: 100, // poll interval in ms, default 250ms
-          simultaneous: 1, // limit to 1 connection per resource at a time
-          timeout: timeout ? timeout * 1000 : timeout, // timeout in ms, default Infinity
-          tcpTimeout: 1000, // tcp timeout in ms, default 300ms
-          window: 1000, // stabilization time in ms, default 750ms
-          strictSSL: false,
-          verbose: false, // force disable verbose logs even if SWA_CLI_DEBUG is enabled
-        });
-        spinner.succeed(`Connected to ${chalk.green(url)} successfully`);
+      const waitOnOneOfResources = hostname === "localhost" ? [`tcp:127.0.0.1:${port}`, `tcp:[::1]:${port}`] : [`tcp:${hostname}:${port}`];
+      let isConnected = false;
+      spinner.start(`Waiting for ${chalk.green(url)} to be ready`);
+      for (let resource of waitOnOneOfResources) {
```

Comment:
Using `for` with `await` is an anti-pattern as it will wait for the first iteration to be resolved before moving on to the second one, not running both promises in parallel (as we want to).

I suggest to refactor it something like this instead:
```js
const promises = waitOnOneOfResources.map(
  resource => {
    return waitOn(...)
      .then(() => resource)
      .catch(err => {
        logger.silly(`Could not connect to ${resource}`);
        throw err;
      });
  }
);
const resource = await Promise.any(promises);
if (resource instanceof AggregateError) {
  // both promises rejected, it failed
} else {
  // resource is connected
}
```

Reply:
The await mthod is modified in the second commit "correct the await method". Thanks for your suggestion.